packages/api/ecs/async-operation/index.js

```diff
@@ -113,14 +115,10 @@ async function fetchLambdaFunction(codeUrl) {
   // of connection timeouts.  If the download fails, this will retry it up to
   // 10 times with an exponential backoff.
   await pRetry(
-    () => new Promise((resolve, reject) => {
-      const file = fs.createWriteStream('/home/task/fn.zip')
-        .on('finish', resolve)
-        .on('error', reject);
-      got.stream(codeUrl)
-        .on('error', reject)
-        .pipe(file);
-    }),
+    () => promisify(pipeline)(
```

Comment:
Based on my reading of `pipeline`, if we promisifying [`pipeline`](https://nodejs.org/api/stream.html#stream_stream_pipeline_source_transforms_destination_callback), then we don't have the callback style error handling, so any error encountered here will just be thrown?

Reply:
That's correct. It's actually a bit safer, since there were potentially events that we _weren't_ handling before. This approach is what is recommended by the `got` documentation.